domain/ide/converter/converter.go

```diff
@@ -52,30 +52,44 @@ func ToCodeActions(issues []snyk.Issue) (actions []lsp.CodeAction) {
 }
 
 func ToCodeAction(issue snyk.Issue, action snyk.CodeAction) lsp.CodeAction {
+	var id *lsp.CodeActionData = nil
```

Comment:
Using pointers here doesn't make sense for me - we don't manipulate data, we just pass it, and thus it should and could use the default "pass-by-value" instead of passing around the pointer. Also, the many stars make it hard to read for a person like me ;)

Reply:
The main issue is nullability. You can't set a value type to `nil`.
I want `Data` to be `nil` when there's no deferred code action to run, so I initialize it to nil and assign it conditionally.